xapi-client/src/main/java/dev/learning/xapi/client/MultipartHelper.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2016-2023 Berry Cloud Ltd. All rights reserved.
+ */
+
+package dev.learning.xapi.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.learning.xapi.model.Attachment;
+import dev.learning.xapi.model.Statement;
+import dev.learning.xapi.model.SubStatement;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.util.FastByteArrayOutputStream;
+import org.springframework.web.reactive.function.client.WebClient.RequestBodySpec;
+
+/**
+ * Helper methods for creating multipart message from statements.
+ *
+ * @author István Rátkai (Selindek)
+ */
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class MultipartHelper {
+
+  private static final String MULTIPART_BOUNDARY = "xapi-learning-dev-boundary";
+  private static final String MULTIPART_CONTENT_TYPE = "multipart/mixed; boundary="
+      + MULTIPART_BOUNDARY;
+  private static final String CRLF = "\r\n";
+  private static final String BOUNDARY_PREFIX = "--";
+  private static final String BODY_SEPARATOR = BOUNDARY_PREFIX + MULTIPART_BOUNDARY + CRLF;
+  private static final String BODY_FOOTER = BOUNDARY_PREFIX + MULTIPART_BOUNDARY + BOUNDARY_PREFIX;
+  private static final String CONTENT_TYPE = HttpHeaders.CONTENT_TYPE + ":";
+
+  private static final byte[] BA_APP_JSON_HEADER = (CONTENT_TYPE + MediaType.APPLICATION_JSON_VALUE
+      + CRLF + CRLF).getBytes(StandardCharsets.UTF_8);
+  private static final byte[] BA_CRLF = CRLF.getBytes(StandardCharsets.UTF_8);
+  private static final byte[] BA_BODY_SEPARATOR = BODY_SEPARATOR.getBytes(StandardCharsets.UTF_8);
+  private static final byte[] BA_BODY_FOOTER = BODY_FOOTER.getBytes(StandardCharsets.UTF_8);
+  private static final byte[] BA_CONTENT_TYPE = CONTENT_TYPE.getBytes(StandardCharsets.UTF_8);
+  private static final byte[] BA_ENCODING_HEADER = ("Content-Transfer-Encoding:binary" + CRLF)
+      .getBytes(StandardCharsets.UTF_8);
+  private static final byte[] BA_X_API_HASH = "X-Experience-API-Hash:"
+      .getBytes(StandardCharsets.UTF_8);
+
+  public static final MediaType MULTIPART_MEDIATYPE = MediaType.valueOf(MULTIPART_CONTENT_TYPE);
+
+  private static final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
I believe this should be injected because the behaviour of the object mapper should be controlled by the application that uses the xAPI Client. The statements without attachments use the spring provided object mapper.